frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -655,6 +655,10 @@ bool TypeInference::canCastBetween(const IR::Type* dest, const IR::Type* src) co
                 return true;
         } else if (dest->is<IR::Type_Boolean>()) {
             return f->size == 1 && !f->isSigned;
+        }  else if (dest->is<IR::Type_Struct>()) {
```

Comment:
I don't see any tests using casts. Can you please add a cast in one of the p4 programs?
Otherwise this code is not exercised.
You will probably need to add a pass to convert such casts to multiple assignments or slices or concatenations.

Reply:
Sure, will do.  Let me also think about any pass.